x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -174,7 +176,13 @@ public Realm realm(String name) {
                 }
                 internalTypes.add(type);
             }
+            if (KerberosRealmSettings.TYPE.equals(type) && registeredRealmTypes.contains(type)) {
+                // Kerberos realm can have only one instance.
+                throw new IllegalArgumentException(
+                        "multiple [" + type + "] realms are configured. For [" + type + "] there can only be one such realm configured");
```

Comment:
Hmm, the more I think about this; this does not feel right to me. Ideally we should have no coupling to a realm implementation in this class. Right now it is just native realms and I'd like to keep it that way for the sole reason of being able to possibly move some things to a separate plugin (like SAML, LDAP, etc); so maybe we can make this a bootstrap check that is always enforced?